rex/resource.py

```diff
@@ -117,6 +114,73 @@ def adder(self):
 
     @staticmethod
     def _check_slice(ds_slice):
+        """
+        Check ds_slice for lists, ensure lists are of the same len
+
+        Parameters
+        ----------
+        ds_slice : int | slice | list | ndarray | tuple
+            What to extract from ds, each arg is for a sequential axis
+
+        Returns
+        -------
+        list_len : int | None
+            List lenght, None if none of the args are a list | ndarray
+        multi_list : bool
+            Flag if multiple list are provided in ds_slice
+        """
+        multi_list = False
+        list_len = []
+        for s in ds_slice:
```

Comment:
The doc string says the input can be an int. Won't this break? Can you iterate on a slice object? same in `_extract_list_slice()`